api/v1alpha1/velero_types.go

```diff
@@ -54,47 +54,70 @@ const CSIPluginImageKey UnsupportedImageKey = "csiPluginImageFqin"
 const ResticRestoreImageKey UnsupportedImageKey = "resticRestoreImageFqin"
 const RegistryImageKey UnsupportedImageKey = "registryImageFqin"
 
-// VeleroSpec defines the desired state of Velero
-type VeleroSpec struct {
-	// BackupStorageLocations defines the list of desired configuration to use for BackupStorageLocations
-	// +optional
-	BackupStorageLocations []velero.BackupStorageLocationSpec `json:"backupStorageLocations"`
-	// VolumeSnapshotLocations defines the list of desired configuration to use for VolumeSnapshotLocations
-	// +optional
-	VolumeSnapshotLocations []velero.VolumeSnapshotLocationSpec `json:"volumeSnapshotLocations"`
-	// VeleroFeatureFlags defines the list of features to enable for Velero instance
+type VeleroConfig struct {
+	// FeatureFlags defines the list of features to enable for Velero instance
 	// +optional
-	VeleroFeatureFlags []string `json:"veleroFeatureFlags,omitempty"`
-	// VeleroTolerations defines the list of tolerations to be applied to Velero deployment
+	FeatureFlags []string `json:"featureFlags,omitempty"`
+	// Tolerations defines the list of tolerations to be applied to Velero deployment
 	// +optional
-	VeleroTolerations []corev1.Toleration `json:"veleroTolerations,omitempty"`
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
```

Comment:
feels to me like Tolerationsp[], ResourceAllocations[], and NodeSelector could be common for both velero deployment as well as restic demonsets. I wonder if it is a good idea to do the following?
```
type PodConfig {
    Tolerations
    ResourceAllocations
    NodeSelector
}

type VeleroConfig struct {
    *PodConfig
    ...
}

type ResticConfig struct {
    *PodConfig
}
```